diff --git a/core/interfaces/src/main/kotlin/app/aaps/core/interfaces/utils/HardLimits.kt b/core/interfaces/src/main/kotlin/app/aaps/core/interfaces/utils/HardLimits.kt
index 4d759863bd..595a5421a7 100644
--- a/core/interfaces/src/main/kotlin/app/aaps/core/interfaces/utils/HardLimits.kt
+++ b/core/interfaces/src/main/kotlin/app/aaps/core/interfaces/utils/HardLimits.kt
@@ -7,16 +7,16 @@ interface HardLimits {
 
         // Very Hard Limits Ranges
         // First value is the Lowest and second value is the Highest a Limit can define
-        val LIMIT_MIN_BG = doubleArrayOf(80.0, 180.0)
-        val LIMIT_MAX_BG = doubleArrayOf(90.0, 200.0)
-        val LIMIT_TARGET_BG = doubleArrayOf(80.0, 200.0)
+        val LIMIT_MIN_BG = doubleArrayOf(72.0, 180.0)
+        val LIMIT_MAX_BG = doubleArrayOf(72.0, 200.0)
+        val LIMIT_TARGET_BG = doubleArrayOf(72.0, 200.0)
 
         // Very Hard Limits Ranges for Temp Targets
         val LIMIT_TEMP_MIN_BG = doubleArrayOf(72.0, 180.0)
         val LIMIT_TEMP_MAX_BG = doubleArrayOf(72.0, 270.0)
         val LIMIT_TEMP_TARGET_BG = doubleArrayOf(72.0, 200.0)
         val MIN_DIA = doubleArrayOf(5.0, 5.0, 5.0, 5.0, 5.0)
-        val MAX_DIA = doubleArrayOf(9.0, 9.0, 9.0, 9.0, 10.0)
+        val MAX_DIA = doubleArrayOf(10.0, 10.0, 10.0, 10.0, 10.0)
         val MIN_IC = doubleArrayOf(2.0, 2.0, 2.0, 2.0, 0.3)
         val MAX_IC = doubleArrayOf(100.0, 100.0, 100.0, 100.0, 100.0)
         const val MIN_ISF = 2.0 // mgdl
